pilot/pkg/bootstrap/istio_ca.go

```diff
@@ -227,23 +230,22 @@ func detectSigningCABundle() (ca.SigningCAFileBundle, error) {
 	tlsSigningFile := path.Join(LocalCertDir.Get(), ca.TLSSecretCACertFile)
 
 	// looking for tls file format (tls.crt)
+	// If tls.crt is used - the cert may include intermediaries, but not the root cert(s)
+	//
 	if _, err := os.Stat(tlsSigningFile); err == nil {
-		log.Info("Using kubernetes.io/tls secret type for signing ca files")
+		log.Infof("Using kubernetes.io/tls secret type for signing ca files %s", LocalCertDir.Get())
 		return ca.SigningCAFileBundle{
-			RootCertFile: path.Join(LocalCertDir.Get(), ca.TLSSecretRootCertFile),
-			CertChainFiles: []string{
-				tlsSigningFile,
-				path.Join(LocalCertDir.Get(), ca.TLSSecretRootCertFile),
-			},
+			RootCertFile:   path.Join(LocalCertDir.Get(), ca.TLSSecretRootCertFile),
+			CertChainFiles: nil,
+			// tls.crt is concatenanted, multiple certs.
 			SigningCertFile: tlsSigningFile,
 			SigningKeyFile:  path.Join(LocalCertDir.Get(), ca.TLSSecretCAPrivateKeyFile),
 		}, nil
 	} else if !os.IsNotExist(err) {
 		return ca.SigningCAFileBundle{}, err
 	}
 
-	log.Info("Using istiod file format for signing ca files")
-	// default ca file format
+	// default ca file format - SigningCertFile holds the leaf CertChain the chain.
```

Comment:
should we check `ca.CertChainFile` exists here

Reply:
I think there is code that checks later - probably we should check, but I'm not sure what to do besides logging something. Do you have any preference ? The roots have a tricky logic - even if the file exists it is not very clear if it will be used.

I would rather break away the code that handles roots - to a clean but separate file that only deals with the roots - and keep the rest of the code handling certs.